master/internal/resourcemanagers/priority.go

```diff
@@ -343,21 +343,19 @@ func taskFilter(label string, zeroSlots bool) func(*sproto.AllocateRequest) bool
 	}
 }
 
-// compareByRegisteredTime sorts tasks by how long their jobs have been submitted.
+// compareByTime compares AllocateRequests by how long their jobs have been submitted.
 // while falling back to when their Allocation actor was created for non-job tasks.
-func compareByRegisteredTime(tasks []*sproto.AllocateRequest, i, j int) bool {
-	aReqI := tasks[i]
-	aReqJ := tasks[j]
-	if aReqI.JobSubmissionTime != nil && aReqJ.JobSubmissionTime != nil {
-		return aReqI.JobSubmissionTime.Before(*aReqJ.JobSubmissionTime)
-	}
-	if aReqI.JobSubmissionTime != nil {
-		return true
-	}
-	if aReqJ.JobSubmissionTime != nil {
+func compareByTime(a *sproto.AllocateRequest, b *sproto.AllocateRequest) bool {
+	switch aReqComparator(a, b) {
+	case 0:
 		return false
+	case 1:
+		return false
+	case -1:
+		return true
+	default:
+		panic("unexpected comparison result")
 	}
```

Comment:
this, and maybe we dont need this func
```suggestion
	return aReqComparator(a, b) < 0
```

Reply:
duh lol

Reply:
I removed it alltogether